src/instrumentation.rs

```diff
@@ -0,0 +1,342 @@
+        timer: &Instant,
+    ) {
+        let duration = timer.elapsed().as_micros() as i64;
+        info!("Record batch completed in {} microseconds", duration);
+        self.record_stat((StatTypes::RecordBatchCompleted, 1)).await;
+        self.record_stat((
+            StatTypes::BufferedRecordBatches,
+            buffered_record_batch_count as i64,
+        ))
+        .await;
+        self.record_stat((StatTypes::RecordBatchWriteDuration, duration))
+            .await;
+    }
+
+    // delta writes
+
+    async fn log_delta_write_started(&self) {
+        info!("Delta write started");
+        self.record_stat((StatTypes::DeltaWriteStarted, 1)).await;
+    }
+
+    async fn log_delta_write_completed(&self, timer: &Instant) {
+        let duration = timer.elapsed().as_micros() as i64;
+        info!("Delta write completed in {} microseconds", duration);
+        self.record_stat((StatTypes::DeltaWriteCompleted, 1)).await;
+        self.record_stat((StatTypes::DeltaWriteDuration, duration))
+            .await;
+    }
+
+    async fn log_delta_write_failed(&self) {
+        info!("Delta write failed");
+        self.record_stat((StatTypes::DeltaWriteFailed, 1)).await;
+    }
+
+    // write ahead log
+
+    async fn log_write_ahead_log_prepared(&self) {
+        info!("Write ahead log entry prepared");
+        self.record_stat((StatTypes::WriteAheadLogEntryPrepared, 1))
+            .await;
+    }
+
+    async fn log_write_ahead_log_completed(&self, timer: &Instant) {
+        let duration = timer.elapsed().as_micros() as i64;
+        info!(
+            "Write ahead log entry completed in {} microseconds",
+            duration
+        );
+        self.record_stat((StatTypes::WriteAheadLogEntryCompleted, 1))
+            .await;
+        self.record_stat((StatTypes::WriteAheadLogEntryDuration, duration))
+            .await;
+    }
+
+    async fn log_write_ahead_log_aborted(&self) {
+        warn!("Write ahead log entry aborted");
+        self.record_stat((StatTypes::WriteAheadLogEntryAborted, 1))
+            .await;
+    }
+
+    // delta tx
+
+    async fn log_delta_tx_version_found(&self, app_id: &str, txn_version: i64) {
+        info!("Read tx version of {} for app {}", txn_version, app_id);
+    }
+
+    async fn log_delta_tx_version_not_found(&self, app_id: &str) {
+        info!(
+            "Delta table does not contain a txn for app id {}. Starting write-ahead-log from 1.",
+            app_id
+        );
+    }
+
+    // control plane
+
+    async fn log_stream_cancelled(&self, m: &BorrowedMessage) {
+        info_message("Found cancellation token set. Stopping run loop.", m);
+    }
+
+    // helpers
+
+    async fn record_stat(&self, statistic: Statistic) {
+        let _ = self.stats_sender().send(statistic).await;
+    }
+
+    fn stats_sender(&self) -> Arc<Sender<Statistic>>;
+}
+
+pub struct StatsHandler {
+    metrics: Arc<StatsdScope>,
+    values: Arc<DashMap<String, i64>>,
```

Comment:
I'm not understanding what the point of this values map is, I am only seeing data going into it, but not coming out of it

Reply:
Yeah - I was thinking about creating a little HTTP endpoint for pulling stats, ultimately decided this was out of scope and forgot to delete the map. 

Removed now.